network.go

```diff
@@ -622,33 +632,41 @@ func (n *network) Delete() error {
 		return &UnknownNetworkError{name: name, id: id}
 	}
 
-	numEps := n.getEpCnt().EndpointCnt()
-	if numEps != 0 {
+	if !force && n.getEpCnt().EndpointCnt() != 0 {
 		return &ActiveEndpointsError{name: n.name, id: n.id}
 	}
 
-	if err = n.deleteNetwork(); err != nil {
-		return err
+	// Mark the network for deletion
+	if !n.inDelete && n.Type() != "host" && n.Type() != "null" {
```

Comment:
Because I thought we were still relying on the driver code:
https://github.com/docker/libnetwork/blob/master/drivers/host/host.go#L41